File: hw1/hw1.tests/CalculatorTests.cs
using System;
using Xunit;

namespace hw1.tests
{
    public class CalculatorTests
    {
        [Theory]
        [InlineData(1, "+", 3, 4)]
        [InlineData(2, "-", 2, 0)]
        [InlineData(3, "*", 1, 3)]
        [InlineData(4, "/", 2, 2)]
        [InlineData(0, "+", 0, 0)]
        [InlineData(1, "/", 1, 1)]
        public void Calculate_ifArgumentsRight(int val1, string operation,
            int val2, int expected)
        {
            var res = Calculator.Calculate(val1, operation, val2);
            
            Assert.Equal(expected, res);
            
        }
    }
}

File: hw5/hw5.Tests/CalculatorTests.fs
module CalculatorTests

open hw5.Calculator
open Xunit

open hw5

[<Theory>]
[<InlineData(1 ,"+", 1, 2)>]
[<InlineData(2 ,"-", 1, 1)>]
[<InlineData(2 ,"*", 2, 4)>]
[<InlineData(2 ,"/", 1, 1)>]

let ``CalculateSuccess`` (arg1 ,operation, arg2, expect) =
    let res = Calculate arg1 operation arg2
    Assert.Equal(res, Some expect)
    
[<Theory>]
[<InlineData(1 ,"/", 0, "DividingByZero")>]
[<InlineData(2 ,"/", 0, "DividingByZero")>]

let ``DividingByZero`` (arg1 ,operation, arg2, expect) =
    
    let res = Calculate arg1 operation arg2
    
    Assert.Equal(res, None expect)

